client/rpc/src/state/mod.rs

```diff
@@ -357,6 +360,15 @@ impl<Block, Client> StateApi<Block::Hash> for State<Block, Client>
 		self.backend.query_storage(from, to, keys)
 	}
 
+	fn query_storage_at(
+		&self,
+		keys: Vec<StorageKey>,
+		at: Option<Block::Hash>
+	) -> FutureResult<Vec<StorageChangeSet<Block::Hash>>> {
+		let at = at.unwrap_or_else(|| self.backend.info().best_hash);
+		self.query_storage(keys,at, Some(at))
```

Comment:
have you considered changing `self.backend.query_storage()` instead of exposing `.info()` method? Alternatively we could even add `self.backend.query_storage_at(at, keys)` as well to keep that contained.